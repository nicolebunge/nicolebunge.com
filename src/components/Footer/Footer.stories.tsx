import Footer from '.';

const meta = {
  component: Footer,
};

const Default = {
  args: {
    children: 'Footer',
  },
};

export { Default };
export default meta;
