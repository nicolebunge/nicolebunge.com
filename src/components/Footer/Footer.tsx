import clsx from 'clsx';
import React, { HTMLAttributes, ReactNode } from 'react';
import { footer } from './Footer.module.css';

interface FooterProps extends HTMLAttributes<HTMLElement> {
  children: ReactNode;
}

function Footer(props: FooterProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <footer className={clsx(className, footer)} role="contentinfo" {...otherProps} />;
}

export type { FooterProps };
export default Footer;
