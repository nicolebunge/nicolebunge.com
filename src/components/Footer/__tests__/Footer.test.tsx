import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import { axe } from 'jest-axe';
import * as stories from '../Footer.stories';

const { Default } = composeStories(stories);

describe('Footer component', () => {
  it('renders correctly', async () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
    expect(await axe(container)).toHaveNoViolations();
  });
});
