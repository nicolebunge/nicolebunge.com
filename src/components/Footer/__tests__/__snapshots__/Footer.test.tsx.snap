// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > renders correctly 1`] = `
<footer
  class="footer"
  role="contentinfo"
>
  Footer
</footer>
`;
